Main.hs

```diff
@@ -208,3 +208,99 @@ updateVector xs n x = runST $ do
   mvector <- V.unsafeThaw xs
   newmvector <- M.write mvector n x
   V.unsafeFreeze mvector
+
+showVal :: LispValue -> String
+showVal (String contents) = "\"" ++ contents ++ "\""
```

Comment:
It's better to use `show contents` here so that control characters would be properly escaped